### DISCUSSION TITLE
[Help] Trigger exported function

### DISCUSSION THREAD

**Discussion by SaschaLucius** [storybookjs/addon-svelte-csf#255] [Q&A] [answered]
Is there a way to trigger an exported function of a component within a Storybook story?

<img width="1020" alt="Image" src="https://github.com/user-attachments/assets/a74cf4ad-2b54-40ac-ada2-16c322a4d743" />
The export got properly documented in Storybook,
but I was not able to find a way to set up a custom control to trigger it.

Thank You

**Comment by xeho91** [comment]
To my knowledge, Storybook doesn't have this kind of control, where you can create a button to emit those events(?). You'd have to define those triggers (e.g., buttons) by yourself. Either inside the component, which is the subject/target of the `*.stories.svelte` file, or inside `<Story>`.

CC @JReinhold Could you confirm that?

What can be done to see those event handlers(?) being _emitted_ is defining [actions](https://storybook.js.org/docs/essentials/actions).

**Comment by JReinhold** [comment]
You have to set up your story to trigger it similar to how it would be triggered in the actual application. Without knowing the internals of the component, how it actually triggers the function, it's hard to say how you could do that.

**Reply by SaschaLucius** [reply]
Thank you for your help @xeho91 and @JReinhold .
I was using the component similar to this [REPL](https://svelte.dev/playground/02cce066556c4b99b159335afa233216?version=5.16.0) and call the clearLog function from the outside.

Is this something I could setup in the story by a button click or insert 2 values to submit the logEvent call?

**Comment by SaschaLucius** [comment, marked as answer]
Hey,
I found a way to interact with the Component.
I was looking for such an way, and was not realizing that the file where I was defining the story, is also a svelte file, which I can use like all other files.

```svelte
<script module>
  import { defineMeta } from '@storybook/addon-svelte-csf';
  import Log from '$lib/Log.svelte';

  // More on how to set up stories at: https://storybook.js.org/docs/writing-stories
  const { Story } = defineMeta({
    title: 'Log',
    component: Log,
    tags: ['autodocs']
  });

  let log = undefined;
</script>

<Story name="Default" args={{ }}/>

<Story name="With Interactions" id="interactions">
  <button on:click={() => log.logEvent('Test')}>Log event</button>
  <button on:click={() => log.logEvent('Test', new Date('2024-01-01'))}>Log event with time</button>
  <button on:click={() => log.clearLog()}>Clear</button>
  <br/>
  <Log bind:this={log}/>
</Story>
```
